src/modules/languages/python.nix

```diff
@@ -63,8 +55,8 @@ let
       # Avoid running "poetry install" for every shell.
       # Only run it when the "poetry.lock" file or python interpreter has changed.
       # We do this by storing the interpreter path and a hash of "poetry.lock" in venv.
-      local ACTUAL_POETRY_CHECKSUM="${cfg.package.interpreter}:$(${pkgs.nix}/bin/nix-hash --type sha256 poetry.lock):''${POETRY_INSTALL_COMMAND[@]}"
-      local POETRY_CHECKSUM_FILE="${venvPath}/poetry.lock.checksum"
+      local ACTUAL_POETRY_CHECKSUM="${cfg.package.interpreter}:$(${pkgs.nix}/bin/nix-hash --type sha256 pyproject.toml):$(${pkgs.nix}/bin/nix-hash --type sha256 poetry.lock):''${POETRY_INSTALL_COMMAND[@]}"
```

Comment:
Won't this fail if pyproject.toml doesn't exist? Maybe that's never the case?

Reply:
The check for pyproject.toml is already here: https://github.com/bobvanderlinden/devenv/blob/fbf8406b622d214890c166676993cf56702a8931/src/modules/languages/python.nix#L80
It won't run any of the poetry functions when it isn't available.